docs/email/postfix/postfix-spf-dkim-debian-8.md

```diff
@@ -0,0 +1,380 @@
+
+    mv YYYYMM.private example.private
+    mv YYYYMM.txt example.txt
+
+Repeat the commands in this step for every entry in the key table. The `-b 2048` indicates the number of bits in the RSA key pair used for signing and verification. 1024 bits is the minimum, but with modern hardware 2048 bits is safer and it's possible 4096 bits will be required at some point.
+
+7.  Make sure the ownership and permissions on `/etc/opendkim` and it's contents are correct by running the following commands:
+
+    cd /etc
+    chown -R opendkim:opendkim /etc/opendkim
+    chmod -R go-rw /etc/opendkim/keys
+
+8.  Check that OpenDKIM starts correctly by:
+
+    systemctl restart opendkim
+
+You should get no error messages from it. If you get errors, use:
+
+    systemctl status -l opendkim
+
+to get the status and untruncated error messages.
+
+### Setting up DNS
+
+As with SPF, DKIM uses TXT records to hold information about the signing key for each domain. Using YYYYMM as above, you need to make a TXT record for the host `YYYYMM._domainkey` for each domain you handle mail for. It's value can be found in the `example.txt` file for the domain. Those files look like this:
+
+{: .file}
+example.txt
+:   ~~~ text
+    201510._domainkey  IN  TXT ( **"v=DKIM1; k=rsa; s=email; "
+        "p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAu5oIUrFDWZK7F4thFxpZa2or6jBEX3cSL6b2TJdPkO5iNn9vHNXhNX31nOefN8FksX94YbLJ8NHcFPbaZTW8R2HthYxRaCyqodxlLHibg8aHdfa+bxKeiI/xABRuAM0WG0JEDSyakMFqIO40ghj/h7DUc/4OXNdeQhrKDTlgf2bd+FjpJ3bNAFcMYa3Oeju33b2Tp+PdtqIwXR"
+        "ZksfuXh7m30kuyavp3Uaso145DRBaJZA55lNxmHWMgMjO+YjNeuR6j4oQqyGwzPaVcSdOG8Js2mXt+J3Hr+nNmJGxZUUW4Uw5ws08wT9opRgSpn+ThX2d1AgQePpGrWOamC3PdcwIDAQAB"** )  ; ----- DKIM key 201510 for example.com
+
+    ~~~
+
+The value inside the parentheses (emphasized above) is what you want. Select the entire region from the double-quote before `v=DKIM1` on to the final double-quote before the closing parentheses. From the above file that would be:
+
+{: .file-excerpt}
+record.txt
+:   ~~~ text
+    "v=DKIM1; k=rsa; s=email; "
+    "p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAu5oIUrFDWZK7F4thFxpZa2or6jBEX3cSL6b2TJdPkO5iNn9vHNXhNX31nOefN8FksX94YbLJ8NHcFPbaZTW8R2HthYxRaCyqodxlLHibg8aHdfa+bxKeiI/xABRuAM0WG0JEDSyakMFqIO40ghj/h7DUc/4OXNdeQhrKDTlgf2bd+FjpJ3bNAFcMYa3Oeju33b2Tp+PdtqIwXR"
+    "ZksfuXh7m30kuyavp3Uaso145DRBaJZA55lNxmHWMgMjO+YjNeuR6j4oQqyGwzPaVcSdOG8Js2mXt+J3Hr+nNmJGxZUUW4Uw5ws08wT9opRgSpn+ThX2d1AgQePpGrWOamC3PdcwIDAQAB"
+    ~~~
+
+It's broken into chunks because of limitations in Bind (one of the most popular DNS server packages) and because of size limitations in the UDP protocol that's usually used for DNS requests and responses. Copy that region and paste it into the value for the TXT record.
+
+If you're using Linode's DNS manager, this is what the add TXT record screen will look like when you have it filled out:
+
+![Linode DNS manager add TXT record](/docs/assets/Postfix_DKIM_TXT_record.png)
```

Comment:
That's referenced up just under "### Setting up DNS", the names for the records will be of the form `YYYYMM._domainkeys` where YYYYMM is the year and month used in step 6 when generating the files. It was 201510 in my example because it was generated in October 2015 and I wanted the example to show what an actual selector would look like rather than a placeholder.